plugins/meta/flannel/flannel_linux.go

```diff
@@ -65,6 +65,12 @@ func doCmdAdd(args *skel.CmdArgs, n *NetConf, fenv *subnetEnv) error {
 		},
 	}
 
+	defer func() {
+		if err != nil {
+			doCmdDel(args, n)
```

Comment:
I think this line will never be hit. We need a `defer` in `delegateAdd`.

Reply:
here: https://github.com/containernetworking/plugins/blob/master/plugins/meta/flannel/flannel.go#L170 ?

add here need check `saveScratchNetConf`, `DelegateAdd`, `return result.Print()`  three cases.

Reply:
no, actually we only need to check `err`